### PR TITLE
Add Yuri

### LINKED ISSUE #144: Yuri mind control
It has to change the owner while a connection is established and the controller isn't dead. http://modenc.renegadeprojects.com/MindControl has some insights.

### PR DESCRIPTION
**Pull request by MustaphaTR**
MC code is by @forcecore, i have modified it a bit tho. PeriodicExplosion (was called ExplodeWeapon on AS) is i think by @GraionDilach.

Currently there are some small problems.

~~1-) It doesn't auto undeploy.~~
2-) MC Arc looks a bit different than original.
3-) MCed unit is missing the ring on it. Couldn't find the correct artwork.
4-) I didn't add the Yuri IFV yet. Someone should start merging the PRs it is hard to work on different branches like that.
5-) Yuri can enter transports when it has something mced. It doesn't break much tho. ~~Just the arc shows last location of Yuri.~~

Closes #144.

### CODE REVIEW

**Comment by DylanC** [comment]
Your mind is clear :)

**Comment by GraionDilach** [comment]
If you're taking AS code to implement Yuri, I'd go with taking GrantTimedConditionOnDeploy along as well and using a ResetReloadWhenEnabled + bigger ReloadDelay on the ExplodeWeapon/PeriodicExplosion weapon than GTCOD duration.

File: OpenRA.Mods.RA2/Traits/MindController.cs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+		[Desc("PipType to use for indicating left over MC capacity")]
+		public readonly PipType PipTypeEmpty = PipType.Transparent;
+
+		public override object Create(ActorInitializer init) { return new MindController(init.Self, this); }
+	}
+
+	class MindController : ConditionalTrait<MindControllerInfo>, INotifyAttack, IPips, INotifyKilled, INotifyActorDisposing, INotifyCreated
+	{
+		readonly MindControllerInfo info;
+		readonly Health health;
```
Comment (Mailaender) [inline OpenRA.Mods.RA2/Traits/MindController.cs]: > Warning as Error: The private field `OpenRA.Mods.RA2.Traits.MindController.health' is assigned but its value is never used

**Comment by GraionDilach** [comment]
GCTOD also might need an update to be in snyc with the changes of https://github.com/AttacqueSuperior/Engine/commit/13850bf69dda17d5141479dbf5535e0da6105cbf#diff-5018706328579f45283ad787eeb089dd.

**Comment by MustaphaTR** [comment]
Updated.

**Comment by Mailaender** [comment]
Deploying yuri crashes

```
Exception of type `System.InvalidOperationException`: Unit `explosion` does not have a sequence named `psi_wave`
  at OpenRA.Graphics.SequenceProvider.GetSequence (System.String unitName, System.String sequenceName) [0x0005e] in <5381d194c6a741728bfd4d9b452f1e2e>:0 
  at OpenRA.Graphics.Animation.GetSequence (System.String sequenceName) [0x0000d] in <5381d194c6a741728bfd4d9b452f1e2e>:0 
  at OpenRA.Graphics.Animation.PlaySequence (System.String sequenceName) [0x00001] in <5381d194c6a741728bfd4d9b452f1e2e>:0 
  at OpenRA.Graphics.Animation.PlayThen (System.String sequenceName, System.Action after) [0x00023] in <5381d194c6a741728bfd4d9b452f1e2e>:0 
  at OpenRA.Mods.Common.Effects.SpriteEffect..ctor (OpenRA.WPos pos, OpenRA.World world, System.String image, System.String sequence, System.String palette, System.Boolean visibleThroughFog, System.Boolean scaleSizeWithZoom, System.Int32 facing) [0x0006c] in <7fec4be7cb654103a8ba5f58a03e3ae0>:0 
  at OpenRA.Mods.Common.Warheads.CreateEffectWarhead+<DoImpact>c__AnonStorey2.<>m__0 (OpenRA.World w) [0x00000] in <7fec4be7cb654103a8ba5f58a03e3ae0>:0 
  at OpenRA.World.Tick () [0x00154] in <5381d194c6a741728bfd4d9b452f1e2e>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x0020a] in <5381d194c6a741728bfd4d9b452f1e2e>:0 
  at OpenRA.Game.LogicTick () [0x00050] in <5381d194c6a741728bfd4d9b452f1e2e>:0 
  at OpenRA.Game.Loop () [0x000d6] in <5381d194c6a741728bfd4d9b452f1e2e>:0 
  at OpenRA.Game.Run () [0x00042] in <5381d194c6a741728bfd4d9b452f1e2e>:0 
  at OpenRA.Program.Run (System.String[] args) [0x00011] in <5381d194c6a741728bfd4d9b452f1e2e>:0 
  at OpenRA.Program.Main (System.String[] args) [0x0004f] in <5381d194c6a741728bfd4d9b452f1e2e>:0
```

probably due to the merge conflict.

**Comment by DylanC** [comment]
@Mailaender - Thanks for reviewing all the PR's on RA2. @MustaphaTR - Thanks for your hard work, much appreciated!

**Comment by GraionDilach** [comment]
>3-) MCed unit is missing the ring on it. Couldn't find the correct artwork.

ra2.mix/conquer.mix/yuricntl.shp

**Comment by LiamKarlMitchell** [comment]
How can I try out yuris revenge in Open RA2?

**Comment by abcdefg30** [comment]
@cookgreen is currently building YR for OpenRA at https://github.com/cookgreen/yr.

**Comment by GraionDilach** [comment]
That projects screams of incompetence though.

**Comment by abcdefg30** [comment]
I'm sure cook appreciates any help in pointing out and fixing issues in case you spot something.

**Comment by Mailaender** [comment]
@forcecore you can collect https://www.bountysource.com/issues/31077424-yuri-mind-control now.

**Pull request merged**